api/v1alpha1/hostedcluster_types.go

```diff
@@ -82,6 +82,11 @@ const (
 	// This is a temporary workaround necessary for compliance reasons on the IBM Cloud side:
 	// no images can be pulled from registries outside of IBM Cloud's official regional registries
 	MachineApproverImage = "hypershift.openshift.io/machine-approver-image"
+
+	// PauseReconciliationAnnotation is an annotation that can be used to pause reconciliation on resources. Either
+	// a date can be provided in RFC3339 format or a boolean. If a date is provided: reconciliation is paused until that date.
+	// If the boolean true is provided it is paused until the annotation is removed.
+	PauseReconciliationAnnotation = "hypershift.openshift.io/pause-reconciliation"
```

Comment:
Should we maybe make this a `spec` field so its easier to discover?

Reply:
I think it's easier from a user perspective to be able to annotate and then remove the annotation from a CLI perspective. I'm open to anything though: other components like cluster-api take the annotation approach as well.

Reply:
Why is it easier? Using the annotation requires knowing the annotation key. If we had a spec field that we do not `omitempty`, you can just `kubectl edit hostedcluster $name`, search for `pause`, set it to `true`

Reply:
I have mixed feelings. Usually I'd vote for something that impacts expected behaviour to be in the API.
However I think the HC API mission is to expose the most meaningful primitives for a consumer to express intent for an end cluster.
So I think the driving factor to expose something in the API should be: does this help a consumer to express intent about its cluster? I think the answer here is no and this is rather plumbing for debuggability, so annotation make sense to me here I think.

Reply:
Well, spec is generally the desired state. The usage-case I imagine is that if I am an SRE and hypershift somehow does the wrong thing, I'd prefer to be able to quickly find this field rather than having to look it up in a doc, which also assumes that I am aware that this exists.

Reply:
edit is not what people will do in jenkins jobs/automation. People will not be doing this by hand:

I do believe it still could be done with a patch command which would be acceptable for automation. However annotate is also just as easy. It would also need to be an easy patch remove command from the resource too.

Reply:
an argument for spec/status is that we can report via condition that the associated controllers are actually paused.  with an annotation alone, dont we have to get under the hood and actually verify the controllers were stopped?

Reply:
I believe we would be able to compute status on either....... but I am open for either or! Mainly just focused on the functionality is all :) 

Basically in the status computation it would have to check for the annotation and if there use it for the status

Reply:
ok let's do spec/status based on Alvaro and Derek feedback

Reply:
refactoring